.github/workflows/release.yml

```
@@ -13,107 +13,240 @@ on:
 
-      - name: Create full ZIP
+      - name: 🔍 Get Release SHA
+        id: set_release_sha
+        run: |
+          RELEASE_SHA=$(git rev-parse ${{ steps.get_release_tag.outputs.tag }})
+          echo "sha=$RELEASE_SHA" >> $GITHUB_OUTPUT
+          echo "Release SHA: $RELEASE_SHA"
+
+      - name: 📦 Create Full Controller Pack ZIP
         run: |
+          mkdir -p UK/Data/Sector
+          echo "${{ steps.set_release_sha.outputs.sha }}" >> UK/Data/Sector/pack_version.txt
           zip -r "uk_controller_pack_${{ steps.get_release_tag.outputs.tag }}.zip" UK/ -x "*.py"
 
-      - name: Upload ZIP as release asset
+      - name: 📤 Upload Full Pack Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: uk_controller_pack_${{ steps.get_release_tag.outputs.tag }}
+          path: uk_controller_pack_${{ steps.get_release_tag.outputs.tag }}.zip
+
+      - name: 🚀 Upload Full ZIP to GitHub Release
         uses: softprops/action-gh-release@v1
         with:
           files: |
             uk_controller_pack_${{ steps.get_release_tag.outputs.tag }}.zip
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
   create-changes-only-zip:
+    needs: create-full-zip
     runs-on: ubuntu-latest
-
+    outputs:
+      changes_tag: ${{ steps.tags.outputs.latest_tag }}
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
+      - name: 👥 Checkout Repository with Full History
+        uses: actions/checkout@v4
         with:
-          fetch-depth: 0 # Fetch all history for accurate comparison
+          fetch-depth: 0
 
-      - name: Get the last release tag using GitHub API
-        id: get_last_release
+      - name: 🏷️ Determine Latest and Previous Release Tags
+        id: tags
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          current_tag=${{ steps.get_release_tag.outputs.tag }}
-          repo=${{ github.repository }}
-          
-          # Get the list of releases using the GitHub API
-          releases=$(curl -s -H "Authorization: token $GITHUB_TOKEN" "https://api.github.com/repos/$repo/releases")
-          
-          # Extract all tags and store them in a sorted array
-          tags=($(echo $releases | jq -r '.[].tag_name' | sort -r))
-          
-          # Find the index of the current tag
-          current_index=$(echo ${tags[@]} | tr ' ' '\n' | grep -n -e "$current_tag" | cut -d: -f1)
-          current_index=$((current_index - 1)) # Convert to 0-based index
-          
-          # Get the previous tag
-          if [ $current_index -eq 0 ]; then
-            echo "This is the first release."
-            previous_tag=""
-          else
-            previous_tag=${tags[$current_index + 1]}
+          repo="${{ github.repository }}"
+          releases=$(curl -s -H "Authorization: token $GITHUB_TOKEN" "https://api.github.com/repos/$repo/releases?per_page=10")
+          tags=($(echo "$releases" | jq -r '.[].tag_name'))
```

@luke11brown - Not sure if this will break stuff, but according to https://github.com/orgs/community/discussions/21901, the api doesn't return them in any order. 
```suggestion
          tags=($(echo "$releases" | jq -r 'sort_by(.published_at) | reverse | .[].tag_name'))
```

something like this might fix it if it becomes a problem